internal/resolver/delegatingresolver/delegatingresolver_test.go

```diff
@@ -0,0 +1,306 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package delegatingresolver
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/grpc/internal/attributes"
+	"google.golang.org/grpc/internal/grpctest"
+	"google.golang.org/grpc/internal/testutils"
+	"google.golang.org/grpc/resolver"
+	_ "google.golang.org/grpc/resolver/dns" // To register dns resolver.
+	"google.golang.org/grpc/resolver/manual"
+	"google.golang.org/grpc/serviceconfig"
+)
+
+type s struct {
+	grpctest.Tester
+}
+
+func Test(t *testing.T) {
+	grpctest.RunSubTests(t, s{})
+}
+
+const (
+	targetTestAddr          = "test.com"
+	resolvedTargetTestAddr  = "1.2.3.4:8080"
+	resolvedTargetTestAddr1 = "1.2.3.5:8080"
+	envProxyAddr            = "proxytest.com"
+	resolvedProxyTestAddr   = "2.3.4.5:7687"
+	resolvedProxyTestAddr1  = "2.3.4.6:7687"
+)
+
+// overwrite function overwrites HTTPSProxyFromEnvironment and
+// returns a function to restore the default values.
+func overwrite(hpfe func(req *http.Request) (*url.URL, error)) func() {
+	originalHPFE := HTTPSProxyFromEnvironment
+	HTTPSProxyFromEnvironment = hpfe
+	return func() {
+		HTTPSProxyFromEnvironment = originalHPFE
+	}
+}
+
+// createTestResolverClientConn initializes a test ResolverClientConn and
+// returns it along with channels for resolver state updates and errors.
+func createTestResolverClientConn(t *testing.T) (*testutils.ResolverClientConn, chan resolver.State, chan error) {
+	stateCh := make(chan resolver.State, 1)
+	errCh := make(chan error, 1)
+
+	tcc := &testutils.ResolverClientConn{
+		Logger:       t,
+		UpdateStateF: func(s resolver.State) error { stateCh <- s; return nil },
+		ReportErrorF: func(err error) { errCh <- err },
+	}
+	return tcc, stateCh, errCh
+}
+
+// TestParsedURLForProxyEnv verifies that the parsedURLForProxy function
+// correctly resolves the proxy URL for a given target address.
+func (s) TestParsedURLForProxyEnv(t *testing.T) {
```

Comment:
I would recommend testing all cases for `parsedURLForProxy`. I'm not asking to set the underlying env vars and checking the result, but instead overriding the func to test at least the following cases:
- a non-nil url and a nil err is returned from the func
- a nil-url and err is returned from the func
- a nil-url and a nil error is returned from the func

you could test all these cases with a table driven test instead of adding one test per scenario.

Reply:
Done.